packages/plugins/src/foundry/utils/index.ts

```diff
@@ -1191,3 +1191,25 @@ export const findFullyQualifiedNameForAddress = (
   //   ? findFullyQualifiedNameForInitCode(initCodeWithArgs, configArtifacts)
   //   : undefined
 }
+
+export const assertSphinxFoundryForkInstalled = async (
+  execAsyncCommand = execAsync
+): Promise<void> => {
+  const { stdout } = await execAsyncCommand(`forge --version`)
+
+  const match = stdout.match(/\((\w+)\s/)
+
+  if (match === null) {
+    throw new Error(`Foundry is not installed.`)
+  }
+
+  const foundryVersion = match[1]
+
+  if (foundryVersion !== 'fcb7c1e') {
```

Comment:
I know you haven't requested a review on this, but we should discuss this further. I don't know if it's practical to use the commit hash to use this because it'll make it quite difficult for us to update the fork with changes from upstream.

Reply:
I think we might want to do some type of empirical check instead, like running script that tests if the state diff works the way that we expect it to work on our fork.